lib/mining/cpuminer.js

```diff
@@ -298,6 +299,37 @@ class CPUMiner extends EventEmitter {
     return [nonce, solved];
   }
 
+  /**
+   * Hash randomly until the nonce overflows.
+   * @param {CPUJob} job
+   * @param {Number} interval
+   * @returns {Array} [nonce, solved]
+   */
+
+  findNonceRandom(job, interval) {
+    const hdr = job.getHeader();
+    const target = job.attempt.target;
+
+    if (!interval)
+      interval = CPUMiner.INTERVAL;
+
+    let nonce = 0;
+    let solved = false;
+
+    for (;;) {
+      nonce = random.randomRange(0, 0xffffffff);
```

Comment:
We should add a constant to `lib/protocol/consensus` for both the size of the nonce and the extra nonce so that we don't have to hardcode in a magic number here